wezterm-gui/src/termwindow/mod.rs

```diff
@@ -928,6 +928,20 @@ impl TermWindow {
                 }
                 Ok(true)
             }
+            WindowEvent::DroppedFile(paths) => {
+                let pane = match self.get_active_pane_or_overlay() {
+                    Some(pane) => pane,
+                    None => return Ok(true),
+                };
+                let paths = paths
+                    .iter()
+                    .map(|path| format!("{:?}", path.to_string_lossy().to_string()))
```

Comment:
I wonder if this is OK?

```suggestion
                    .map(|path| format!("{}", path.display()))
```

Reply:
Cannot achieve such an effect like https://github.com/wez/wezterm/pull/1868#discussion_r849133789 , is still an unescaped string without quotes. :(

Reply:
OK, looking at iterm2, when dropping filenames with spaces, what it does it backslash escape any spaces.  So I think what we should do here is something like:

```rust
    .map(|path| path.to_string_lossy().replace(" ", "\\ "))
```

Reply:
Not just spaces, there are other characters that need to be escaped.
![image](https://user-images.githubusercontent.com/8097526/163746658-0c7eeeaf-a719-4289-848d-7765b6015e28.png)
Why not consider  outputting a string enclosed in quotes?
![image](https://user-images.githubusercontent.com/8097526/163746890-1827ab69-a988-49d8-9648-aac053468312.png)

Reply:
> Why not consider outputting a string enclosed in quotes?

@wez Any ideas? like kitty:
![image](https://user-images.githubusercontent.com/8097526/164877789-9531e238-540f-4d53-9f8e-99756a97243c.png)

Reply:
A problem with quotes is that quoting rules are different between unix shells and windows systems, and we may not know which system we are interacting with.

This sounds like a good place to introduce a config option:

```rust
#[derive(Debug, Deserialize, Serialize, Clone, Copy, PartialEq, Eq)]
enum DroppedFileQuoting {
  /// Do no quoting at all, just pass through the name as-is
  None,
  /// Backslash escape only spaces, leaving all other characters as-is
  SpacesOnly,
  /// Use POSIX style shell word escaping
  Posix,
}

impl Default for DroppedFileQuoting {
  fn default() -> Self {
    Self::SpacesOnly
  }
}

impl DroppedFileQuoting {
   fn escape(s: &str) -> String {
      match self {
         Self::None => s.to_string(),
         Self::SpacesOnly => s.replace(" ", "\\ "),
         // https://docs.rs/shlex/latest/shlex/fn.quote.html
         Self::Posix => shlex.quote(s).into_owned().to_string(),
      }
   }
}
```

and then add this to the `Config` struct in `config/src/config.rs`:

```rust
   #[serde(default)]
   quote_dropped_files: DroppedFileQuoting,
```

Then the code here in this PR would map the value something like this;

```rust
   .map(|path| config.quote_dropped_files.escape(path.to_string_lossy()))
```

Reply:
Good idea!